src/Components/Web/src/Routing/NavLink.cs

```diff
@@ -124,9 +132,24 @@ private bool ShouldMatch(string currentUriAbsolute)
             return true;
         }
 
+        if (_disableMatchAllIgnoresLeftUriPart)
+        {
+            return false;
+        }
+
+        string uriWithoutQueryAndFragment = GetUriIgnoreQueryAndFragment(currentUriAbsolute);
+        if (Match == NavLinkMatch.All
+            && EqualsHrefExactlyOrIfTrailingSlashAdded(uriWithoutQueryAndFragment))
```

Comment:
Don't we also need to trim the query string from `_hrefAbsolute`?

Reply:
I think that the expectation was to match `NavLink` to the current uri, not the other way around.
However, you're right that not making it symmetric might add some confusion, so for simplicity, we should trim both.